chain/vm/fvm.go

```diff
@@ -13,20 +19,209 @@ import (
 	ffi "github.com/filecoin-project/filecoin-ffi"
 	ffi_cgo "github.com/filecoin-project/filecoin-ffi/cgo"
 
+	"github.com/filecoin-project/lotus/chain/actors/adt"
+	"github.com/filecoin-project/lotus/chain/actors/builtin/miner"
 	"github.com/filecoin-project/lotus/chain/types"
 	"github.com/ipfs/go-cid"
 )
 
 var _ VMI = (*FVM)(nil)
+var _ ffi_cgo.Externs = (*FvmExtern)(nil)
 
 type FvmExtern struct {
 	Rand
 	blockstore.Blockstore
+	epoch   abi.ChainEpoch
+	lbState LookbackStateGetter
+	base    cid.Cid
+}
+
+// Similar to the one in syscalls.go used by the Lotus VM, except it never errors
+// Errors are logged and "no fault" is returned, which is functionally what go-actors does anyway
+func (x *FvmExtern) VerifyConsensusFault(ctx context.Context, a, b, extra []byte) *ffi_cgo.ConsensusFaultWithGas {
+	ret := &ffi_cgo.ConsensusFaultWithGas{
+		// TODO: is this gonna be a problem on the receiving end? should we return address.NewIDAddress(0) instead?
+		Target: address.Undef,
```

Comment:
Probably? I'd consider just returning `nil` in this case, honestly. We can cook up the correct FFI return values in the glue code.

Reply:
kk